cmd/metrics-server/app/options/options.go

```diff
@@ -41,51 +38,19 @@ type Options struct {
 	Authentication *genericoptions.DelegatingAuthenticationOptions
 	Authorization  *genericoptions.DelegatingAuthorizationOptions
 	Features       *genericoptions.FeatureOptions
+	KubeletClient  *KubeletClientOptions
 
-	Kubeconfig string
+	MetricResolution time.Duration
+	ShowVersion      bool
+	Kubeconfig       string
 
 	// Only to be used to for testing
 	DisableAuthForTesting bool
-
-	MetricResolution time.Duration
-
-	KubeletUseNodeStatusPort     bool
-	KubeletPort                  int
-	InsecureKubeletTLS           bool
-	KubeletPreferredAddressTypes []string
-	KubeletCAFile                string
-	KubeletClientKeyFile         string
-	KubeletClientCertFile        string
-
-	ShowVersion bool
-
-	DeprecatedCompletelyInsecureKubelet bool
 }
 
 func (o *Options) Validate() []error {
 	errors := []error{}
-	if (o.KubeletCAFile != "") && o.InsecureKubeletTLS {
-		errors = append(errors, fmt.Errorf("Cannot use both --kubelet-certificate-authority and --kubelet-insecure-tls"))
-	}
-
-	if (o.KubeletClientKeyFile != "") != (o.KubeletClientCertFile != "") {
-		errors = append(errors, fmt.Errorf("Need both --kubelet-client-key and --kubelet-client-certificate"))
-	}
-
-	if (o.KubeletClientKeyFile != "") && o.DeprecatedCompletelyInsecureKubelet {
-		errors = append(errors, fmt.Errorf("Cannot use both --kubelet-client-key and --deprecated-kubelet-completely-insecure"))
-	}
-
-	if (o.KubeletClientCertFile != "") && o.DeprecatedCompletelyInsecureKubelet {
-		errors = append(errors, fmt.Errorf("Cannot use both --kubelet-client-certificate and --deprecated-kubelet-completely-insecure"))
-	}
-
-	if o.InsecureKubeletTLS && o.DeprecatedCompletelyInsecureKubelet {
-		errors = append(errors, fmt.Errorf("Cannot use both --kubelet-insecure-tls and --deprecated-kubelet-completely-insecure"))
-	}
-	if (o.KubeletCAFile != "") && o.DeprecatedCompletelyInsecureKubelet {
-		errors = append(errors, fmt.Errorf("Cannot use both --kubelet-certificate-authority and --deprecated-kubelet-completely-insecure"))
-	}
+	errors = append(errors, o.KubeletClient.Validate()...)
```

Comment:
you can return ` o.KubeletClient.Validate()...` directly here

Reply:
done